tests/copy/with-parents.earth

```diff
@@ -0,0 +1,45 @@
+VERSION --copy-parent-dirs 0.6
```

Comment:
Not sure how we would write a test to validate the files we sent to the build context, but here's some manual validation I did as suggested by @alexcb :

```
./t/copy+copy-with-parents |          alpine:3.13 | --> Load metadata linux/amd64
./t/copy+copy-with-parents |          alpine:3.13 | Completed in 2.008680026s
./t/copy+copy-with-parents |              context | --> local context .
./t/copy+copy-with-parents |              context | Completed in 37.11µs
./t/copy+copy-with-parents |              context | --> local context .
./t/copy+copy-with-parents |              context | Completed in 21.29µs
./t/copy+copy-with-parents |                +base | --> FROM alpine:3.13
./t/copy+copy-with-parents |                +base | [██████████] 100% resolve docker.io/library/alpine:3.13@sha256:2582893dec6f12fd499d3a709477f2c0c0c1dfcd28024c93f1f0626b9e3540c8
./t/copy+copy-with-parents |                +base | Completed in 1.540602ms
./t/copy+copy-with-parents |                +base | Completed in 27.25µs
./t/copy+copy-with-parents |                +base | Completed in 376.805µs
./t/copy+copy-with-parents |              context | [          ]   0% transferring ../..:
./t/copy+copy-with-parents |              context | sent file stat for ../test1
./t/copy+copy-with-parents |              context | sent file stat for ../test1/foo1
./t/copy+copy-with-parents |              context | sent data for ../test1/foo1 (5 B)
./t/copy+copy-with-parents |              context | [██████████] 100% transferring ..:
./t/copy+copy-with-parents |              context | Completed in 1.555302ms
./t/copy+copy-with-parents |              context | sent file stat for ../../opt
./t/copy+copy-with-parents |              context | sent file stat for ../../opt/test2
./t/copy+copy-with-parents |              context | sent file stat for ../../opt/test2/foo2
./t/copy+copy-with-parents |              context | sent data for ../../opt/test2/foo2 (5 B)
./t/copy+copy-with-parents |                +base | [          ]   0% sha256:4e9f2cdf438714c2c4533e28c6c41a89cc6c1b46cf77e54c488db30ca4f5b6f3
./t/copy+copy-with-parents |              context | [██████████] 100% transferring ../..:
./t/copy+copy-with-parents |              context | Completed in 170.921911ms

```

Reply:
> Not sure how we would write a test to validate the files we sent to the build context, but here's some manual validation I did as suggested by @alexcb :

Maybe we can extend https://github.com/earthly/earthly/blob/main/tests/Earthfile#L168-L178 to include copying a file from a parent directory?

Reply:
Ah great catch, let me add that

Reply:
updated